src/App.tsx

```diff
@@ -16,35 +22,101 @@
 ];
 
 export const App: React.FC = () => {
+  const [sort, setSort] = useState<string[]>([...goodsFromServer]);
+  const [reverse, setReverse] = useState<boolean>(false);
+  const [active, setActive] = useState<string>('');
+
+  const handleSort = (good: Cases): void => {
+    let newSort: string[];
+
+    switch (good) {
+      case Cases.Abc:
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.localeCompare(a));
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.localeCompare(b));
+          setActive(good);
+        }
+
+        break;
+      case Cases.Length:
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.length - a.length);
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.length - b.length);
+          setActive(good);
+        }
+
+        break;
+      case Cases.Reverse:
+        if (active === Cases.Abc || active === Cases.Length) {
+          setReverse(!reverse);
+        } else if (active === Cases.Reverse) {
+          setReverse(!reverse);
+          setActive('');
+        } else {
+          setReverse(!reverse);
+          setActive(Cases.Reverse);
+        }
+
+        newSort = [...sort].reverse();
+        break;
+      default:
+        setActive('');
+        setReverse(false);
+        newSort = [...goodsFromServer];
+    }
+
+    setSort(newSort);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${(active !== 'abc' ? 'is-light' : '') || (active !== 'abc' && reverse) ? 'is-light' : ''}`}
+          onClick={() => handleSort(Cases.Abc)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${(active !== 'length' ? 'is-light' : '') || (active !== 'length' && reverse) ? 'is-light' : ''}`}
```

Comment:
Similar to the previous button, the className logic here is also complex and might not work as intended. Simplifying this logic would be beneficial.